dot/core/service_test.go

```diff
@@ -546,24 +559,32 @@ func Test_Service_maintainTransactionPool(t *testing.T) {
 			Propagate: true,
 		}
 
-		extrinsic := types.Extrinsic{21}
-		vt := transaction.NewValidTransaction(extrinsic, validity)
-		tx := transaction.NewValidTransaction(types.Extrinsic{21}, &transaction.Validity{Propagate: true})
+		ext := types.Extrinsic{21}
+		externalExt := types.Extrinsic(append([]byte{byte(types.TxnExternal)},
+			append(ext, testHeader.StateRoot.ToBytes()...)...))
+		vt := transaction.NewValidTransaction(ext, validity)
+		tx := transaction.NewValidTransaction(ext, &transaction.Validity{Propagate: true})
 
 		ctrl := gomock.NewController(t)
 		runtimeMock := new(mocksruntime.Instance)
-		runtimeMock.On("ValidateTransaction", types.Extrinsic{21}).
+		runtimeMock.On("ValidateTransaction", externalExt).
 			Return(&transaction.Validity{Propagate: true}, nil)
+		runtimeMock.On("Version").Return(runtimeVersionData, nil)
 		mockTxnState := NewMockTransactionState(ctrl)
 		mockTxnState.EXPECT().RemoveExtrinsic(types.Extrinsic{21})
 		mockTxnState.EXPECT().PendingInPool().Return([]*transaction.ValidTransaction{vt})
 		mockTxnState.EXPECT().Push(tx).Return(common.Hash{}, nil)
 		mockTxnState.EXPECT().RemoveExtrinsicFromPool(types.Extrinsic{21})
 		mockBlockStateOk := NewMockBlockState(ctrl)
 		mockBlockStateOk.EXPECT().GetRuntime(nil).Return(runtimeMock, nil)
+		mockBlockStateOk.EXPECT().BestBlockHash().Return(common.Hash{})
+		mockStorageState := NewMockStorageState(ctrl)
+		mockStorageState.EXPECT().Lock()
+		mockStorageState.EXPECT().Unlock()
```

Comment:
nit a bit of a paranoid approach, but checking the order of calls can be nice especially for locking:

```suggestion
		lockCall := mockStorageState.EXPECT().Lock()
		mockStorageState.EXPECT().Unlock().After(lockCall)
```

Same for other mock expectations of lock/unlock in this test file